docs/modules/ROOT/pages/architecture/traits.adoc

```diff
@@ -1,6 +1,65 @@
+
+NOTE: This document reflects Camel K version 1.5. It may not reflect slight changes developed after this review.
+
+== Traits life cycle
+
+Traits are typically used to tune several aspects of an `Integration`. However, you will learn that we are using the same concept to influence the build of `IntegrationKits`. Therefore, we can distinguish between those traits that can be applied to either one or the other type. Another important thing to know is that the platform uses this mechanism to perform many common (hidden to the user) operations. We use to identify those trait as **platform traits**. Misusing a platform trait, may result in execution errors.
+
+Another important concept related to the trait lifecycle is the **trait profile**. At this time, Camel K is supporting the following profiles:
+
+* Kubernetes
+* Openshift
+* Knative
+
+A profile is useful to identify on which kind of cluster a trait has to run: vanilla `Kubernetes`, `Openshift` or OpenShift/Kubernetes clusters powered by `Knative`. The default is to allow a trait on any profile; each trait can specify a different behavior (ie, running a trait only for a profile as it happens with `Knative`).
+
+=== Trait configuration
+
+A Camel K user will provide a trait configuration via CLI (`--trait` or `-t` flag) or setting directly its configuration into the `Integration` resource. Each trait has a unique id and can expose any kind of parameter ie, `kamel run -t [trait-id].[key]=[value]`. The operator will transform that _key_ and _value_ in the related trait variable.
+
+As an example, let's see what to do when you want to set an additional `JVM` classpath pointing to a known dependency on your `Integration`:
+
+[source,console]
+----
+$ kamel run --trait jvm.classpath=/path/to/my.jar MyIntegration.java
+----
+
+=== Trait interface
+
+In order to understand better the logic behind the trait managment, let's have a look at the `https://github.com/apache/camel-k/blob/main/pkg/trait/trait_types.go#L70[Trait]` interface:
+
+[source,go]
+----
+type Trait interface {
+	Identifiable
+	client.Injectable
+	InjectContext(context.Context)
+	Configure(environment *Environment) (bool, error)
+	Apply(environment *Environment) error
+	InfluencesKit() bool
+	IsPlatformTrait() bool
+	RequiresIntegrationPlatform() bool
+	IsAllowedInProfile(v1.TraitProfile) bool
+	Order() int
+}
+----
+
+Each trait will implement this interface. The most important functions that will be invoked by the xref:architecture/operator.adoc[Operator] are `Configure()` and `Apply()`. Basically, the `Configure()` function will set those inputs aforementioned (each trait has its own). The function is in charge to verify also the correctness of those expected parameters, where it makes sense (ie, a well expected `Kubernetes` resource name).
```

Comment:
functions -> methods

Reply:
I guess in these case they are methods as they are related to behavior of the types that will implement that interface, right?

Reply:
Right, I think method is the right terminology for interfaces, as they can be attached to types: https://golang.org/doc/effective_go#interface_methods.